### PR TITLE
Adjust trailing space when format a segment

### PR DESCRIPTION
**Pull request by juliaroldi**
When select a sentence or word with trailing spaces at the end, split the segment in two. One for the text and other for the trailing spaces and apply the formatting only in the text segment. 
splitTrailingSpacesElements
splitTrailingSpaces

### CODE REVIEW

**Comment by BryanValverdeU** [comment]
When the selection is only spaces format APIs are not applying format to the selection. Just checking if this is desired.

**Comment by juliaroldi** [comment]
> When the selection is only spaces format APIs are not applying format to the selection. Just checking if this is desired.
> 
> I am able to repro with this snapshot and selecting the spaces at the end.
> 
> ```
> <div>Select spaces after -&gt;&nbsp; &nbsp; &nbsp; &nbsp;</div>
> ```

Add a fix for this scenario

File: packages-content-model/roosterjs-content-model-editor/lib/publicApi/utils/adjustTrailingSpaceSelection.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    paragraph: ContentModelParagraph | null,
+    isTrailingSpace: boolean
+): [ContentModelSegment | null, ContentModelParagraph | null] {
+    if (segment && segment.segmentType == 'Text' && paragraph) {
+        const text = segment.text.trimRight();
+        const trailingSpacing = segment.text.substring(text.length);
+        if (text && trailingSpacing) {
+            segment.text = text;
+            const trailingSpacingSegment = createText(trailingSpacing);
+            trailingSpacingSegment.format = {};
```
Comment (JiuqingSong) [inline packages-content-model/roosterjs-content-model-editor/lib/publicApi/utils/adjustTrailingSpaceSelection.ts:20]: why empty the format? It should have the same format with original text?

File: packages-content-model/roosterjs-content-model-editor/lib/publicApi/utils/adjustTrailingSpaceSelection.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            paragraph.segments.push(trailingSpacingSegment);
+        } else {
+            const trailingSpacingSegment = paragraph.segments[paragraph.segments.length - 1];
+            if (isTrailingSpace) {
+                trailingSpacingSegment.format = {};
```
Comment (JiuqingSong) [inline packages-content-model/roosterjs-content-model-editor/lib/publicApi/utils/adjustTrailingSpaceSelection.ts:25]: same, why?

File: packages-content-model/roosterjs-content-model-editor/lib/publicApi/utils/formatSegmentWithContentModel.ts [hunk 3]
```diff
@@ -80,3 +89,19 @@ export function formatSegmentWithContentModel(
...
+
+const isATrailingSpace = (
+    segment: ContentModelSegment | null,
+    paragraph: ContentModelParagraph | null
+) => {
+    if (segment && paragraph && segment.segmentType === 'Text') {
+        return (
+            segment.text.trim().length === 0 &&
+            paragraph.segments.length > 0 &&
+            paragraph.segments[paragraph.segments.length - 1] === segment &&
```
Comment (JiuqingSong) [inline packages-content-model/roosterjs-content-model-editor/lib/publicApi/utils/formatSegmentWithContentModel.ts:101]: need null check

File: packages-content-model/roosterjs-content-model-editor/lib/publicApi/utils/formatSegmentWithContentModel.ts [hunk 4]
```diff
@@ -80,3 +89,19 @@ export function formatSegmentWithContentModel(
         }
     });
 }
+
+const isATrailingSpace = (
```
Comment (JiuqingSong) [inline packages-content-model/roosterjs-content-model-editor/lib/publicApi/utils/formatSegmentWithContentModel.ts:93]: Move this into adjustTrailingSpaceSelection.ts

**Pull request closed without merging**